packages/alfa-rules/src/common/predicate/is-clipped.ts

```diff
@@ -64,6 +64,52 @@ function isClippedBySize(
           }
         }
       }
+
+      if (x.value === "auto" || y.value === "auto") {
+        const { value: height } = style.computed("height");
+        const { value: width } = style.computed("width");
+
+        for (const dimension of [height, width]) {
+          switch (dimension.type) {
+            case "percentage":
+              if (dimension.value <= 0) {
+                return true;
+              } else {
+                break;
+              }
+
+            case "length":
+              if (dimension.value <= 1) {
+                return true;
+              } else {
+                break;
+              }
+          }
+        }
+      }
+
+      if (x.value === "scroll" || y.value === "scroll") {
+        const { value: height } = style.computed("height");
+        const { value: width } = style.computed("width");
```

Comment:
Also, since `height` and `width` are used in several places (also in the `"hidden"` case), we can refactor the definitions out of the `if`.